api/src/main/java/org/openmrs/validator/VisitValidator.java

```diff
@@ -107,58 +111,80 @@ public void validate(Object target, Errors errors) {
 				validateStopDatetime(visit, otherVisit, errors);
 			}
 		}
+		Date startDate = visit.getStartDatetime();
+		Patient patient = visit.getPatient();
+		Date birthDate;
+		if(patient!=null) {
+			birthDate = patient.getBirthdate();
+			if(startDate!=null && birthDate!=null) {
+				//Comparing patient's birthDate with their startDate to make sure that DOB is before startDate
+  			if(OpenmrsUtil.compare(startDate,birthDate) < 0) {
+					errors.rejectValue("startDatetime","Visit.error.startDateBeforeBirthDate");
+				}
+				else {
+					// Get the difference between the dates to check if the patient is at least 1 year old
+					double diffInDays = ((startDate.getTime() - birthDate.getTime()) / (1000 * 60 * 60 * 24 ));
```

Comment:
You have misunderstood the logic that is supposed to be carried out here. Sorry if my proposal on the ticket was not clear. I will comment on TRUNK-4939.